hackathon_site/hardware/serializers.py

```diff
@@ -0,0 +1,41 @@
+from rest_framework import serializers
+from hardware.models import Hardware, Category, Order
+
+
+class HardwareSerializer(serializers.ModelSerializer):
+    quantity_remaining = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Hardware
+        fields = (
+            "id",
+            "name",
+            "model_number",
+            "manufacturer",
+            "datasheet",
+            "quantity_available",
+            "notes",
+            "max_per_team",
+            "picture",
+            "categories",
+            "quantity_remaining",
+        )
+
+    def get_quantity_remaining(self, obj):
+        return (
+            obj.quantity_available
+            - Order.objects.filter(items__hardware__id=obj.id).count()
+        )
+
+
+class CategorySerializer(serializers.ModelSerializer):
+    hardware = HardwareSerializer(many=True, read_only=True)
```

Comment:
This wasn't in our spec, and we don't need it here anyway. Once filters are added to the Hardware endpoint, we can filter them by category that way. This will end up just being extra computational and response payload overhead.